### PR TITLE
storage: don't include RaftTombstoneKey in consistency checks

### PR DESCRIPTION
**Pull request by petermattis**
See #12130

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/12131)

### CODE REVIEW

**Comment by petermattis** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/storage/store.go, line 3025 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131#-KYP-mcW4KTcD1BRDzPE:-KYP-mcW5ROO_NxEWQcj:bjryfyu) ([raw file](https://github.com/cockroachdb/cockroach/blob/feb1921c9bb2787c66a8f662f8555d784771f337/pkg/storage/store.go#L3025)):*
> ```Go
> 				log.Warningf(ctx, "unable to retrieve tombstone: %s", err)
> 			} else if ok {
> 				log.Fatalf(ctx, "tombstone should not exist")
> ```

This check fires in `TestRaftRemoveRace`. Not sure if we should keep it or try to move a similar check into that test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r1, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/storage/store.go, line 3025 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131#-KYP-mcW4KTcD1BRDzPE:-KYPjxxl8gLXkU6uoT5C:b26nyl5) ([raw file](https://github.com/cockroachdb/cockroach/blob/feb1921c9bb2787c66a8f662f8555d784771f337/pkg/storage/store.go#L3025)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

This check fires in `TestRaftRemoveRace`. Not sure if we should keep it or try to move a similar check into that test.
</blockquote></details>

What did you conclude here? Why did this check fire here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131)*

**Comment by petermattis** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/storage/store.go, line 3025 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131#-KYP-mcW4KTcD1BRDzPE:-KYPklBeEkuWfC13TY6k:bc8ewn4) ([raw file](https://github.com/cockroachdb/cockroach/blob/feb1921c9bb2787c66a8f662f8555d784771f337/pkg/storage/store.go#L3025)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

What did you conclude here? Why did this check fire here?
</blockquote></details>

The check fires when `raft.IsEmptySnap(ready.Snapshot)` is true. In that case `applySnapshot()` is a no-op and doesn't touch the existing replica data leaving the tombstone that was just added in place. Hence the change above to not add the tombstone key and only to bump `Replica.mu.minReplicaID`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131)*

**Comment by tamird** [comment]
:lgtm: though the method setTombstoneKey is mildly detritus-y now.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131#-:-KYPlloL-ncKSZX_wnA6:b-b6tlw3)*

File: pkg/storage/replica_command.go [hunk 1]
```diff
@@ -2005,11 +2005,22 @@ func (r *Replica) sha512(
...
 	iter := NewReplicaDataIterator(&desc, snap, true /* replicatedOnly */)
 	defer iter.Close()
+
 	for ; iter.Valid(); iter.Next() {
 		key := iter.Key()
+		if key.Equal(tombstoneKey) {
+			// Skip the tombstone key which is marked as replicated even though it
+			// isn't.
+			//
+			// TODO(peter): Figure out a way to migrate this key to the unreplicated
```
Comment (spencerkimball) [inline pkg/storage/replica_command.go]: Did you file an issue for this? Seems problematic and worth fixing sooner than a TODO in the code is likely to prompt.

**Comment by bdarnell** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12131#-:-KYScd0IJSvpIldF90DN:bnfp4nl)*

**Pull request merged**